tests/test_switch.py

```diff
@@ -584,6 +584,146 @@ def increased_color_temp():
+    for entity_id in [ENTITY_SWITCH, ENTITY_SLEEP_MODE_SWITCH]:
+        await change_manual_control(True)
+        assert manual_control[ENTITY_LIGHT]
+        await turn_switch(False, entity_id)
+        await turn_switch(True, entity_id)
+        assert not manual_control[ENTITY_LIGHT]
+
+    # Check that when 'adapt_brightness' is off, changing the brightness
+    # doesn't mark it as manually controlled but changing color_temp
+    # does
+    await turn_light(False)  # reset manually controlled status
+    await turn_light(True)
+    assert not manual_control[ENTITY_LIGHT]
+    await switch.adapt_brightness_switch.async_turn_off()
+    await turn_light(True, brightness=increased_brightness())
+    assert not manual_control[ENTITY_LIGHT]
+    mired_range = (light.min_color_temp_kelvin, light.max_color_temp_kelvin)
+    kelvin_range = (
+        color_temperature_mired_to_kelvin(mired_range[1]),
+        color_temperature_mired_to_kelvin(mired_range[0]),
+    )
+    ptp_kelvin = kelvin_range[1] - kelvin_range[0]
+    await turn_light(True, color_temp_kelvin=(light._ct + 100) % ptp_kelvin)
+    assert manual_control[ENTITY_LIGHT]
+    await switch.adapt_brightness_switch.async_turn_on()  # turn on again
+
+    # Check that when 'adapt_color' is off, changing the color
+    # doesn't mark it as manually controlled but changing brightness
+    # does
+    await turn_light(False)  # reset manually controlled status
+    await turn_light(True)
+    assert not manual_control[ENTITY_LIGHT]
+    await switch.adapt_color_switch.async_turn_off()
+    await turn_light(True, color_temp=increased_color_temp())
+    assert not manual_control[ENTITY_LIGHT]
+    await turn_light(True, brightness=increased_brightness())
+    assert manual_control[ENTITY_LIGHT]
+
+    # Check that when 'adapt_color' adapt_brightness are both off
+    # nothing marks it as manually controlled
+    await turn_light(False)  # reset manually controlled status
+    await turn_light(True)
+    await switch.adapt_color_switch.async_turn_off()
+    await switch.adapt_brightness_switch.async_turn_off()
+    assert not manual_control[ENTITY_LIGHT]
+    await turn_light(True, color_temp=increased_color_temp())
+    await turn_light(True, brightness=increased_brightness())
+    await turn_light(
+        True,
+        color_temp=increased_color_temp(),
+        brightness=increased_brightness(),
+    )
+    assert not manual_control[ENTITY_LIGHT]
+    # Turn switches on again
+    await switch.adapt_color_switch.async_turn_on()
+    await switch.adapt_brightness_switch.async_turn_on()
+
+    # Check that when no lights are specified, all are reset
+    await change_manual_control(True, {CONF_LIGHTS: switch._lights})
+    assert all([manual_control[eid] for eid in switch._lights])
+    # do not pass "lights" so reset all
+    await change_manual_control(False, {})
+    assert all([not manual_control[eid] for eid in switch._lights])
```

Comment:
I just noticed that this is a 100% carbon copy of the `test_manual_control` test. Why did you add this? This debugging?